benchto-driver/src/main/java/com/teradata/benchto/driver/graphite/GraphiteClient.java

```diff
@@ -108,8 +108,18 @@ private URI buildLoadMetricsURI(Map<String, String> metrics, ZonedDateTime from,
         double[] dataPoints = new double[inputDataPoints.length];
         for (int i = 0; i < inputDataPoints.length; i++) {
             Double value = inputDataPoints[i][DATA_POINT_VALUE_INDEX];
-            // rarely graphite does not return value for given timestamp, use 0 in this case
-            dataPoints[i] = value == null ? 0 : value;
+            if (value == null) {
+                /*
+                 * Graphite returns null for an aggregation if *all* ingredients are null. We should query Graphite after
+                 * delay, so that all ingredients *are* present, so aggregation should not be null too.
+                 *
+                 * Note however, that non-null value for e.g. an average or sum does not mean that the value can be trusted.
+                 * Some data points can still be missing.
+                 */
+                LOGGER.error("null data point returned from Graphite");
```

Comment:
I think it is better to do not have a metric than metric which is misleading. So once again, if it is known that there is value missing I would fail here (and retry later for couple of times?)